tests/test_basic.py

```diff
@@ -105,3 +106,70 @@ def test_basic_transform_resize(capsys, test_data):
     basic.darkfield = np.full(basic.flatfield.shape, 8)
     corrected = basic.transform(images)
     assert corrected.mean() == corrected_error
+
+
+def test_basic_save_model(tmp_path: Path):
+    model_dir = tmp_path / "test_model"
+
+    basic = BaSiC()
+
+    # set profiles
+    basic.flatfield = np.full((128, 128), 1, dtype=np.float64)
+    basic.darkfield = np.full((128, 128), 2, dtype=np.float64)
+
+    # save the model
+    basic.save_model(model_dir)
+
+    # check that the files exists
+    assert (model_dir / "settings.json").exists()
+    assert (model_dir / "profiles.npy").exists()
+
+    # load files and check for expected content
+    saved_profiles = np.load(model_dir / "profiles.npy")
+    profiles = np.dstack((basic.flatfield, basic.darkfield))
+    assert np.array_equal(saved_profiles, profiles)
+
+    # TODO check settings contents
+
+    # an error raises when the model folder exists
+    with pytest.raises(FileExistsError):
+        basic.save_model(model_dir)
+
+
+def test_basic_load_model(tmp_path: Path):
+
+    model_dir = tmp_path / "test_model"
+
+    # raises error when folder not found
+    with pytest.raises(FileNotFoundError):
+        basic = BaSiC.load_model(tmp_path)
```

Comment:
Let's add an input parameter to test this.

https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions

Reply:
What would the input parameter be for?

Reply:
A pytedt parameter that passes in a non existent model path. We can mark it as an expected failure.

Reply:
It looks like `@pytest.mark.parametrize` doesn't like fixtures as parameters. I can parametrize fixtures themselves, but then the fixtures will have to return a tuple that is a path (real or fake) and the boolean of whether there is an expected error. I'll give it a shot.